File: model-client/src/main/java/org/modelix/model/api/IWriteTransaction.java
package org.modelix.model.api;

public interface IWriteTransaction extends ITransaction {
  void setProperty(long nodeId, String role, String value);
  void setReferenceTarget(long sourceId, String role, INodeReference target);
  void moveChild(long newParentId, String newRole, int newIndex, long childId);
  long addNewChild(long parentId, String role, int index, IConcept concept);
  void addNewChild(long parentId, String role, int index, long childId, IConcept concept);
  void deleteNode(long nodeId);

  void setTree(ITree tree);
}

File: model-client/src/main/kotlin/org/modelix/model/api/IWriteTransaction.kt
package org.modelix.model.api

interface IWriteTransaction : ITransaction {
    fun setProperty(nodeId: Long, role: String?, value: String?)
    fun setReferenceTarget(sourceId: Long, role: String?, target: INodeReference?)
    fun moveChild(newParentId: Long, newRole: String?, newIndex: Int, childId: Long)
    fun addNewChild(parentId: Long, role: String?, index: Int, concept: IConcept?): Long
    fun addNewChild(parentId: Long, role: String?, index: Int, childId: Long, concept: IConcept?)
    fun deleteNode(nodeId: Long)
    //fun setTree(tree: ITree?)
}